packages/pnpm/test/recursive/run.ts

```diff
@@ -157,6 +157,116 @@ test('`pnpm recursive run` fails if none of the packaegs has the desired command
   }
 })
 
+test('`pnpm recursive run` fails when run with a filter that includes all packages and no package has the desired command', async (t: tape.Test) => {
+  const projects = preparePackages(t, [
+    {
+      name: 'project-1',
+      version: '1.0.0',
+
+      dependencies: {
+        'json-append': '1',
+      },
+      scripts: {
+        build: `node -e "process.stdout.write('project-1')" | json-append ../output.json`,
+      },
+    },
+    {
+      name: 'project-2',
+      version: '1.0.0',
+
+      dependencies: {
+        'json-append': '1',
+        'project-1': '1'
+      },
+      scripts: {
+        build: `node -e "process.stdout.write('project-2')" | json-append ../output.json`,
+      },
+    },
+    {
+      name: 'project-3',
+      version: '1.0.0',
+
+      dependencies: {
+        'json-append': '1',
+        'project-1': '1'
+      },
+      scripts: {
+        build: `node -e "process.stdout.write('project-3')" | json-append ../output.json`,
```

Comment:
these builds scripts are not used in this test, so remove them and also remove the json-append deps